app/models/piece.rb

```diff
@@ -12,9 +12,9 @@ def self.color
 
   def move_to!(destination_x, destination_y)
     save_piece_capturable_by_en_passant(destination_x, destination_y)
-    destination_piece = game.pieces.find_by(column_coordinate: destination_x, row_coordinate: destination_y, is_on_board?: true)
-    raise 'Invalid Move' if destination_piece.present? && !capturable?(destination_piece)
-    if en_passant_move?(destination_x, destination_y)
+    destination_piece = find_destination_piece(destination_x, destination_y)
+    raise 'Invalid Move' unless capturable?(destination_piece)
+    if type == 'Pawn' && en_passant_move?(destination_x, destination_y)
```

Comment:
This works. An even better way to do it would be to have `Pawn` own this logic.

You could override the `move_to!` method inside of Pawn. Then specifically when you call `move_to!` on a Pawn, it will use the logic inside of Pawn, rather than Piece.

Another thing you can do, when overriding a method. Look up what `super` does. You can take advantage of existing logic in the parent class. 😄